src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -119,6 +119,34 @@ public void Dispose()
         public void WriteUInt16(ushort value) => _writer.Write(value);
         public void WriteString(string value) => WriteStringValue(value);
 
+        /// <summary>
+        /// Used so we can easily grab the low/high 64bits of a guid for serialization.
+        /// </summary>
+        [StructLayout(LayoutKind.Explicit)]
+        internal struct GuidAccessor
+        {
+            unsafe static GuidAccessor()
+            {
+                Debug.Assert(sizeof(Guid) == 16);
```

Comment:
:bulb: This is simply unnecessary (no way it could ever fail). If any size is off, `TestRoundTripGuid` will fail for the case of a random GUID.